### PR TITLE
fixed issue #2084 where password reset from email does not verify user's email

### PR DESCRIPTION
**Pull request by prdpx7**
this commit fixes issue #2084

### CODE REVIEW

File: allauth/account/forms.py [hunk 1]
```diff
@@ -556,6 +556,14 @@ def __init__(self, *args, **kwargs):
 
     def save(self):
         get_adapter().set_password(self.user, self.cleaned_data["password1"])
+        try:
+            # verify user's EmailAddress on password reset via email
+            email_address = EmailAddress.objects.get_for_user(self.user, user_email(self.user))
```
Comment (pennersr) [inline allauth/account/forms.py]: A user can have multiple email addresses. Here, you happen to return the one that is one user.email, but it is not 100% certain that this was the email address the password reset mail was sent to.

File: allauth/account/forms.py [hunk 2]
```diff
@@ -556,6 +556,14 @@ def __init__(self, *args, **kwargs):
...
+            email_address.verified = True
+            email_address.save()
```
Comment (pennersr) [inline allauth/account/forms.py]: `.save(update_fields=['verified'])`

**Comment by pennersr** [comment]
Closing -- see my comments.

**Comment by danlamanna** [comment]
@pennersr I'm not the original author, but I'm interested in this quality of life feature.

It seems like placing the email in the token itself is a strict requirement for this feature. If a PR were to be made that had some sort of `EmailBasedPasswordResetTokenGenerator` which instead of being a user id and timestamp were an email address and timestamp, would you be open to accepting it?

**Pull request closed without merging**